libsolidity/codegen/ir/IRGenerationContext.h

```diff
@@ -154,6 +252,29 @@ class IRGenerationContext
 	/// all platforms - which is a property guaranteed by MultiUseYulFunctionCollector.
 	std::set<FunctionDefinition const*> m_functionGenerationQueue;
 
+	/// Collection of functions that need to be callable via internal dispatch.
+	/// These are internal functions which satisfy all of the following conditions:
+	/// 1. Are referenced by name in an expression other than a direct function call.
+	/// 2. There exists at least one call of any internal function of the same arity via a pointer.
```

Comment:
Do need that condition during code creation? Such a dispatch routine should be removed by the optimizer.

Reply:
I didn't want to assume that it will be removed by the optimizer. But if I can make this assumption then yeah, just generating dispatches corresponding to all detected functions will make the logic much simpler.

Reply:
Done. This simplified the code a lot.